components/prism-avisynth.js

```diff
@@ -0,0 +1,193 @@
+		// conditional
+		/conditional(?:filter|select|reader)|frameevaluate|scriptclip|writefile(?:if|start|end)?|animate|applyrange|tcp(?:server|source)/.source,
+		// export
+		/imagewriter/.source,
+		// debug
+		/subtitle|blankclip|blackness|colorbars(?:hd)?|compare|dumpfiltergraph|setgraphanalysis|echo|histogram|info|messageclip|preroll|showfiveversions|show(?:framenumber|smpte|time)|stack(?:horizontal|vertical)|tone|version/.source
+	].join('|');
+	var allinternals = [internals, properties, filters].join('|');
+
+	Prism.languages.avisynth = {
+		'comment': [
+			{
+				// Matches [* *] nestable block comments, but only supports 1 level of nested comments
+				// /\[\*(?:[^\[*]|\[(?!\*)|\*(?!\])|<self>)*\*\]/
+				pattern: /(^|[^\\])\[\*(?:[^\[*]|\[(?!\*)|\*(?!\])|\[\*(?:[^\[*]|\[(?!\*)|\*(?!\]))*\*\])*\*\]/,
+				lookbehind: true,
+				greedy: true
+			},
+			{
+				// Matches /* */ block comments
+				pattern: /(^|[^\\])\/\*[\s\S]*?(?:\*\/|$)/,
+				lookbehind: true,
+				greedy: true
+			},
+			{
+				// Matches # comments
+				pattern: /(^|[^\\$])#.*/,
+				lookbehind: true,
+				greedy: true
+			}
+		],
+
+		// Handle before strings because optional arguments are surrounded by double quotes.
+		'argument': {
+			pattern: re(/\b(?:<<0>>)\s+("?)\w+\1/.source, [types], 'i'),
+			inside: {
+				'keyword': /^\w+/
+			}
+		},
+
+		// Optional argument assignment
+		'argument-label': {
+			pattern: /([,(][\s\\]*)\w+\s*=(?!=)/,
+			lookbehind: true,
+			inside: {
+				'argument-name': {
+					pattern: /^\w+/,
+					alias: 'punctuation'
+				},
+				'punctuation': /=$/
+			}
+		},
+
+		'string': [
+			{
+				// triple double-quoted
+				pattern: /"""[\s\S]*?"""/,
+				greedy: true,
+			},
+			{
+				// single double-quoted
+				pattern: /"(?:\\(?:\r\n|[\s\S])|[^"\\\r\n])*"/,
+				greedy: true,
+				inside: {
+					'constant': {
+						// These *are* case-sensitive!
+						pattern: /\b(?:DEFAULT_MT_MODE|(?:SCRIPT|MAINSCRIPT|PROGRAM)DIR|(?:USER|MACHINE)_(?:PLUS|CLASSIC)_PLUGINS)\b/
+					}
+				}
+			}
+		],
+
+		// The special "last" variable that takes the value of the last implicitly returned clip.
+		'variable': /\b(?:last)\b/i,
+
+		'boolean': /\b(?:true|false|yes|no)\b/i,
+
+		'keyword': {
+			pattern: re(/((?:^|[\\{])\s*)(?:<<0>>)(?=\s)/.source, [keywords], 'im'),
+			lookbehind: true
+		},
+
+		'constant': /\bMT_(?:NICE_FILTER|MULTI_INSTANCE|SERIALIZED|SPECIAL_MT)\b/,
+
+		// AviSynth's internal functions, filters, and properties
+		'builtin-function': {
+			pattern: re(/\b(?:<<0>>)\b(?!\s*=)/.source, [allinternals], 'i'),
+			alias: 'function'
+		},
+
+		// External/user-defined filters, and type casts.
+		'function': {
+			pattern: /\b[a-z_]\w*(?=\s*\()|(\.)[a-z_]\w*\b/i,
+			lookbehind: true,
+			inside: {
+				'keyword': {
+					// type casts
+					pattern: re(/\b(?:<<0>>)\b/.source, [types], 'i')
```

Comment:
Variables/parameters **can** be named after type names, as I have mentioned in my comment below. I did realize though that the typecast pattern no longer fits in the external function pattern since `.type` is not valid, so I have separated it out.

Reply:
Your new approach looks good.